app/models/plugins/thermometer.rb

```diff
@@ -30,65 +40,35 @@ def name
     self.class.name.demodulize
   end
 
-  def update_goal
-    increment!(:goal, determine_next_goal - goal)
-  end
-
   private
 
+  def action_count
+    @action_count ||= if page.campaign_id.present?
+      Page.where(campaign_id: page.campaign_id).sum(:action_count)
+    else
+      page.action_count
+    end
+  end
+
   def abbreviate_number(number)
     return number.to_s if number < 1000
     return "#{(goal / 1000).to_i}k" if number < 1_000_000
     locale = page.try(:language).try(:code)
     return "%g #{I18n.t('thermometer.million', locale: locale)}" % (goal / 1_000_000.0).round(1)
   end
 
-  def goal_should_update
-    current_total >= goal
-  end
-
-  def determine_next_goal
-    target_jump = determine_target_jump(self.current_total)
-
-
-    # This is slight overkill; it makes sure that when we increment the goal, it's targeting the closest jump value.
-    # Essentially, the goal here is to make sure that if we somehow don't update the goal right when it's on the next
-    # step (like 200 or 10000) we don't end up with an ugly goal like 301.
-    new_goal = current_total + target_jump
-    goal_target_difference = new_goal % target_jump
-    target_midpoint = target_jump * 0.5
 
-    if goal_target_difference == 0
-      new_goal
-    elsif goal_target_difference <= target_midpoint
-      new_goal - goal_target_difference
-    else
-      new_goal + target_jump - (goal_target_difference)
+  def dynamic_goals
+    Enumerator.new do |y|
+      new_goal = GOALS.last
+      loop {
```

Comment:
`loop` control structure is pretty cool, I like it, although is not that common. I try to avoid doing hacky stuff just because it's cool, but I think an Enumerator was the perfect abstraction for this use case. 
About the curly braces vs do-end, I tend to use curly braces even on short multi-line blocks to help denote hierarchy. However, this is a very personal styling taste thing, I'm happy to follow the standard of always using do-end in multi line blocks.